src/workspace/listLocalProjects.ts

```diff
@@ -0,0 +1,119 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { IActionContext, callWithTelemetryAndErrorHandling } from "@microsoft/vscode-azext-utils";
+import * as vscode from "vscode";
+import { FuncVersion, tryParseFuncVersion } from "../FuncVersion";
+import { tryGetFunctionProjectRoot } from "../commands/createNewProject/verifyIsProject";
+import { getFunctionAppName, getJavaDebugSubpath } from "../commands/initProjectForVSCode/InitVSCodeStep/JavaInitVSCodeStep";
+import { JavaBuildTool, ProjectLanguage, funcVersionSetting, javaBuildTool, projectLanguageModelSetting, projectLanguageSetting } from "../constants";
+import { localize } from "../localize";
+import { dotnetUtils } from "../utils/dotnetUtils";
+import { getWorkspaceSetting } from "../vsCodeConfig/settings";
+import { LocalProject, WorkspaceProject } from "./LocalProject";
+import path = require("path");
+
+interface UnitializedLocalProject {
+    workspaceFolder: vscode.WorkspaceFolder;
+    projectPath: string;
+}
+
+interface InvalidLocalProject extends UnitializedLocalProject {
+    error: unknown;
+}
+
+interface ListWorkspaceProjectsResult {
```

Comment:
It's not immediately obvious to me that these are all exclusive arrays. I think it'd be better to rename it to `initalizedProjects` or maybe have another array that _is_ all of the projects, regardless if it is invalid or uninitalized.